distribution/lib/Standard/Base/0.0.0-dev/src/Data.enso

```diff
@@ -91,11 +93,16 @@ read : Text | URI | File -> File_Format -> Problem_Behavior -> Any ! File_Error
 read path=(Missing_Argument.throw "path") format=Auto_Detect (on_problems : Problem_Behavior = ..Report_Warning) = case path of
     _ : Text -> if Data_Read_Helpers.looks_like_uri path then Data_Read_Helpers.fetch_following_data_links path format=format else
         read (File.new path) format on_problems
-    uri : URI -> fetch uri format=format
+    uri : URI ->
+        response = fetch uri format=format
+        Telemetry.log "Data.read" "fetching from URI" (Dictionary.from_vector [["content_length", response.content_length]])
+        response
     _ ->
         file_obj = File.new path
         if file_obj.is_directory then Error.throw (Illegal_Argument.Error "Cannot `read` a directory, use `Data.list`.") else
-            file_obj.read format on_problems
+            file_content = file_obj.read format on_problems
+            Telemetry.log "Data.read" "read from file" (Dictionary.from_vector [["size", file_obj.size]])
```

Comment:
- log MIME type
- log extension
- type of `file_content`

Reply:
Added extension in b9ca2398514f9f8e980689ffe1415e5d6f83542e. All the other telemetry collection shall be added by the libs team in upcoming PRs. Collection size and file_extension is enough for demonstration.